clinica/pydra/statistics_volume/task.py

```diff
@@ -0,0 +1,633 @@
+    from clinica.utils.spm import spm_standalone_is_available
+
+    assert isinstance(m_file, str), "[Error] Argument must be a string"
+    if not isfile(m_file):
+        raise FileNotFoundError("[Error] File " + m_file + "does not exist")
+    assert m_file[-2:] == ".m", (
+        "[Error] " + m_file + " is not a Matlab file (extension must be .m)"
+    )
+
+    # Generate command line to run
+    if spm_standalone_is_available():
+        utls.delete_last_line(m_file)
+        # SPM standalone must be run directly from its root folder
+        if platform.system().lower().startswith("darwin"):
+            # Mac OS
+            cmdline = (
+                "cd $SPMSTANDALONE_HOME && ./run_spm12.sh $MCR_HOME batch " + m_file
+            )
+        elif platform.system().lower().startswith("linux"):
+            # Linux OS
+            cmdline = "$SPMSTANDALONE_HOME/run_spm12.sh $MCR_HOME batch " + m_file
+        else:
+            raise SystemError("Clinica only support Mac OS and Linux")
+        system(cmdline)
+    else:
+        MatlabCommand.set_default_matlab_cmd(get_matlab_command())
+        matlab = MatlabCommand()
+        if platform.system().lower().startswith("linux"):
+            matlab.inputs.args = "-nosoftwareopengl"
+        matlab.inputs.paths = dirname(m_file)
+        matlab.inputs.script = basename(m_file)[:-2]
+        matlab.inputs.single_comp_thread = False
+        matlab.inputs.logfile = abspath("./matlab_output.log")
+        matlab.run()
+    output_mat_file = abspath(join(dirname(m_file), "..", "2_sample_t_test", "SPM.mat"))
+    if not isfile(output_mat_file):
+        raise RuntimeError("Output matrix " + output_mat_file + " was not produced")
+    return output_mat_file
+
+
+def delete_last_line(filename):
+    """
+    Use this function to remove the call to spm jobman if m file is used with SPM standalone
+    Args:
+        filename: path to filename
+
+    Returns:
+        Nothing
+    """
+    import os
+
+    with open(filename, "r+", encoding="utf-8") as file:
+
+        # Move the pointer (similar to a cursor in a text editor) to the end of the file
+        file.seek(0, os.SEEK_END)
+
+        # This code means the following code skips the very last character in the file -
+        # i.e. in the case the last line is null we delete the last line
+        # and the penultimate one
+        pos = file.tell() - 1
+
+        # Read each character in the file one at a time from the penultimate
+        # character going backwards, searching for a newline character
+        # If we find a new line, exit the search
+        while pos > 0 and file.read(1) != "\n":
+            pos -= 1
+            file.seek(pos, os.SEEK_SET)
+
+        # So long as we're not at the start of the file, delete all the characters ahead
+        # of this position
+        if pos > 0:
+            file.seek(pos, os.SEEK_SET)
+            file.truncate()
```

Comment:
I'm a little bit confused by the complexity and low level look of this function while its name is `delete_last_line`.
Did you have a look at what it does? If it really just removes the last line of a file, I think we can find much simpler ways to do it...